libpromises/evalfunction.c

```diff
@@ -4860,23 +4861,48 @@ static FnCallResult FnCallFormat(EvalContext *ctx, ARG_UNUSED const Policy *poli
 
 static FnCallResult FnCallIPRange(EvalContext *ctx, ARG_UNUSED const Policy *policy, ARG_UNUSED const FnCall *fp, const Rlist *finalargs)
 {
+    const bool iprange_mode = (strcmp(fp->name, "iprange") == 0);
     char *range = RlistScalarValue(finalargs);
 
     if (!FuzzyMatchParse(range))
     {
         return FnFailure();
     }
 
-    for (const Item *ip = EvalContextGetIpAddresses(ctx); ip != NULL; ip = ip->next)
+    /* Case 1: iprange(): No second argument, just arg1: check arg1 (a range) against VIPADDRESS */
+    if (iprange_mode && finalargs->next == NULL)
     {
-        Rlist *ifaces = finalargs->next;
-        // we match on VIPADDRESS iff no interfaces were requested
-        if (FuzzySetMatch(range, VIPADDRESS) == 0 && NULL == ifaces)
+        if (FuzzySetMatch(range, VIPADDRESS) == 0)
+        {
+            Log(LOG_LEVEL_DEBUG, "%s: found range %s on hostip %s, no interface", fp->name, range, VIPADDRESS);
+            return FnReturnContextTrue();
+        }
+
+        return FnReturnContextFalse();
+    }
+
+    /* Case 2: isipinsubnet(): check arg1 (a range) against arg2, arg3, ... if they are IPs */
+    for (Rlist *ip = finalargs->next; !iprange_mode && ip != NULL; ip = ip->next)
```

Comment:
I'm confused, why are you checking `iprange_mode` on every iteration of the loop? Same comment for the next loop.